python/pyiceberg/table/__init__.py

```diff
@@ -773,11 +779,11 @@ def plan_files(self) -> Iterable[FileScanTask]:
         data_entries: List[ManifestEntry] = []
         positional_delete_entries = SortedList(key=lambda entry: entry.data_sequence_number or INITIAL_SEQUENCE_NUMBER)
 
-        with ThreadPool() as pool:
+        with DynamicExecutor() as executor:
```

Comment:
I think this is awesome. I have one more suggestion. What do you think of being able to set the executor through the config?

Reply:
`None` doesn't seem to work on my end:
```
export PYICEBERG_CONCURRENCY_MODE=None   
python3 /tmp/vo.py               
Traceback (most recent call last):
  File "/tmp/vo.py", line 1, in <module>
    from pyiceberg.catalog import load_catalog
  File "/Users/fokkodriesprong/Desktop/iceberg/python/pyiceberg/catalog/__init__.py", line 42, in <module>
    from pyiceberg.serializers import ToOutputFile
  File "/Users/fokkodriesprong/Desktop/iceberg/python/pyiceberg/serializers.py", line 26, in <module>
    from pyiceberg.table.metadata import TableMetadata, TableMetadataUtil
  File "/Users/fokkodriesprong/Desktop/iceberg/python/pyiceberg/table/__init__.py", line 72, in <module>
    from pyiceberg.utils.concurrent import DynamicExecutor
  File "/Users/fokkodriesprong/Desktop/iceberg/python/pyiceberg/utils/concurrent.py", line 146, in <module>
    concurrency_mode = _concurrency_mode()
  File "/Users/fokkodriesprong/Desktop/iceberg/python/pyiceberg/utils/concurrent.py", line 143, in _concurrency_mode
    raise ValueError(f"Invalid concurrency mode: {mode}")
ValueError: Invalid concurrency mode: None
```